### PR TITLE
Implement storage for Group and GroupList

### PR DESCRIPTION
**Pull request by ivan24004**

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -69,29 +73,44 @@ public void init() throws Exception {
...
+            initialContactList = new ContactList();
         }
+        Optional<ReadOnlyGroupList> groupListOptional;
+        ReadOnlyGroupList initialGroupList;
 
-        return new ModelManager(initialData, userPrefs, new GroupList());
+        try {
+            groupListOptional = storage.readGroupList();
+            if (!groupListOptional.isPresent()) {
+                logger.info("Creating a new data file " + storage.getGroupListFilePath()
```
Comment (benson1029) [inline src/main/java/seedu/address/MainApp.java:103]: Do we have a sample group list?

File: src/main/java/seedu/address/storage/GroupListStorage.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import seedu.address.commons.exceptions.DataLoadingException;
+import seedu.address.model.GroupList;
+import seedu.address.model.ReadOnlyGroupList;
+
+/**
+ * Represents a storage for {@link GroupList}
```
Comment (benson1029) [inline src/main/java/seedu/address/storage/GroupListStorage.java:12]: Missing full stop
```suggestion
 * Represents a storage for {@link GroupList}.
```

File: src/main/java/seedu/address/storage/JsonAdaptedGroup.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted group object into the model's {@code Group} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted group.
+     */
+    public Group toModelType() throws IllegalValueException {
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()));
```
Comment (benson1029) [inline src/main/java/seedu/address/storage/JsonAdaptedGroup.java:51]: A long line. Consider breaking it into two.

**Review by benson1029 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/MainApp.java (hunk 1 above)
Comment (ivan24004) [reply src/main/java/seedu/address/MainApp.java:103]: Removed line regarding sampleGroupList

Comment (benson1029) [reply src/main/java/seedu/address/MainApp.java:103]: Great, thanks!

File: src/main/java/seedu/address/storage/JsonAdaptedGroup.java (hunk 3 above)
Comment (ivan24004) [reply src/main/java/seedu/address/storage/JsonAdaptedGroup.java]: Okay, broken in two

**Pull request merged**